pkg/alert/alert.go

```diff
@@ -24,149 +24,44 @@ package alert
-		return nil, fmt.Errorf("failed to create pod: %v", err)
-	}
-	deployment.AddPod(pod)
-
-	return deployment, nil
+// SpecConfig will contain the specification of Alert
+type SpecConfig struct {
+	config *v1.AlertSpec
 }
 
-func (a *SpecConfig) alertPod() (*components.Pod, error) {
-	pod := components.NewPod(horizonapi.PodConfig{
-		Name: "alert",
-	})
-	pod.AddLabels(map[string]string{"app": "alert", "tier": "alert"})
-
-	pod.AddContainer(a.alertContainer())
-
-	vol, err := a.alertVolume()
-	if err != nil {
-		return nil, fmt.Errorf("error creating volumes: %v", err)
-	}
-	pod.AddVolume(vol)
-
-	return pod, nil
+// NewAlert will create the Alert object
+func NewAlert(config *v1.AlertSpec) *SpecConfig {
+	return &SpecConfig{config: config}
 }
 
-func (a *SpecConfig) alertContainer() *components.Container {
-	// This will prevent it from working on openshift without a privileged service account.  Remove once the
-	// chowns are removed in the image
-	user := int64(0)
-	container := components.NewContainer(horizonapi.ContainerConfig{
-		Name:   "alert",
-		Image:  fmt.Sprintf("%s/%s/%s:%s", a.config.Registry, a.config.ImagePath, a.config.AlertImageName, a.config.AlertImageVersion),
-		MinMem: a.config.AlertMemory,
-		UID:    &user,
-	})
-
-	container.AddPort(horizonapi.PortConfig{
-		ContainerPort: "8443",
-		Protocol:      horizonapi.ProtocolTCP,
-	})
-
-	container.AddVolumeMount(horizonapi.VolumeMountConfig{
-		Name:      "dir-alert",
-		MountPath: "/opt/blackduck/alert/alert-config",
-	})
-
-	container.AddEnv(horizonapi.EnvConfig{
-		Type:     horizonapi.EnvFromConfigMap,
-		FromName: "alert",
-	})
-
-	container.AddLivenessProbe(horizonapi.ProbeConfig{
-		ActionConfig: horizonapi.ActionConfig{
-			Command: []string{"/usr/local/bin/docker-healthcheck.sh", "https://localhost:8443/alert/api/about"},
-		},
-		Delay:           240,
-		Timeout:         10,
-		Interval:        30,
-		MinCountFailure: 5,
-	})
-
-	return container
-}
+// GetComponents will return the list of components for alert
+func (a *SpecConfig) GetComponents() (*api.ComponentList, error) {
+	components := &api.ComponentList{}
 
-func (a *SpecConfig) alertVolume() (*components.Volume, error) {
-	vol, err := components.NewEmptyDirVolume(horizonapi.EmptyDirVolumeConfig{
-		VolumeName: "dir-alert",
-		Medium:     horizonapi.StorageMediumDefault,
-	})
+	// Add alert components
+	components.Deployments = append(components.Deployments, a.getAlertDeployment())
+	components.Services = append(components.Services, a.getAlertService())
+	components.Services = append(components.Services, a.getAlertExposedService())
+	components.ConfigMaps = append(components.ConfigMaps, a.getAlertConfigMap())
+	pvc, err := a.getAlertPersistentVolumeClaim()
 	if err != nil {
-		return nil, fmt.Errorf("failed to create empty dir volume: %v", err)
+		return nil, fmt.Errorf("failed to get alert components: %s", err)
+	}
+	components.PersistentVolumeClaims = append(components.PersistentVolumeClaims, pvc)
```

Comment:
Instead of forcing us to use persistent volume, can you implement similar to Black Duck?

Reply:
Alert checks a spec field PersistentStorage like Blackduck does when creating a PersistentVolumeClaim. If false it creates EmptyDir. (It defaults to true)